Source/JavaScriptCore/runtime/ProgramExecutable.cpp

```diff
@@ -287,9 +287,9 @@ void ProgramExecutable::visitChildrenImpl(JSCell* cell, Visitor& visitor)
     ProgramExecutable* thisObject = jsCast<ProgramExecutable*>(cell);
     ASSERT_GC_OBJECT_INHERITS(thisObject, info());
     Base::visitChildren(thisObject, visitor);
-    if (TemplateObjectMap* map = thisObject->m_templateObjectMap.get()) {
+    if (auto& map = thisObject->m_templateObjectMap; !map.isEmpty()) {
```

Comment:
Is this thread safe? If the JS mutator is resizing this right as the marker thread looks at the map this could UAF. I think this should be `!map.isNullStorage()`

Reply:
I don't see how the old code would have been thread safe if the new code isn't. There was still a hash table pointer, just one step away.

Reply:
I'll undo this part; I think we could use a thread-safe boolean for this optimization instead of checking the status of the hash table with a tricky use of a write barrier, but for now I'll just roll this part back to the old way.

Reply:
On second thought I could use `isNullStorage` as you suggest and it should be no less thread safe than the `unique_ptr` version. I guess I'll do that.